sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -58,3 +98,185 @@ async fn test_get_blob_properties_invalid_container() -> Result<(), Box<dyn Erro
 
     Ok(())
 }
+
+#[recorded::test]
+async fn test_download_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let mut options = BlobClientOptions::default();
+    recording.instrument(&mut options.client_options);
+
+    // Setup
+    let endpoint = format!(
+        "https://{}.blob.core.windows.net/",
+        recording.var("AZURE_STORAGE_ACCOUNT_NAME", None).as_str()
+    );
+
+    // Act
+    let container_client = ContainerClient::new(
+        &endpoint,
+        String::from("testcontainer2"),
+        recording.credential(),
+        Some(options.clone()),
+    )?;
+    container_client.create_container(None).await?;
+
+    let blob_client = BlobClient::new(
+        &endpoint,
+        String::from("testcontainer2"),
+        String::from("test_blob.txt"),
+        recording.credential(),
+        Some(options),
+    )?;
+    let data = b"test download content";
+    blob_client
+        .upload_blob(
+            RequestContent::from(data.to_vec()),
+            true,
+            i64::try_from(data.len())?,
+            None,
+        )
+        .await?;
+    let response = blob_client
+        .download_blob(Some(BlobBlobClientDownloadOptions::default()))
+        .await?;
+
+    // Assert
+    let (status_code, headers, response_body) = response.deconstruct();
```

Comment:
At this point, we have a `response_body` of type [ResponseBody](https://github.com/Azure/azure-sdk-for-rust/blob/main/sdk/typespec/typespec_client_core/src/http/response.rs#L274), is there any examples on how to directly stream from this type? While this test currently does test functionality and verifies the content, I would like to be able to test the more common customer scenario (i.e. streaming directly from the download stream to a file or different workflow without buffering all into memory).

Reply:
It's already a stream, and we're actually going to be changing that to buffer the entire stream within the pipeline, but will have a way - for these upload/download methods - to opt out of that. Keep in mind, though, like most (all, probably) other Azure SDK languages, that currently won't be part of the pipeline. I talked with several other architects and those languages don't retry streams like that but agree they should - probably a `RetriableStream` that wraps the pipeline.

As for an example, see https://github.com/Azure/azure-sdk-for-rust/blob/b39649ba194c49326811c5a7cad5ab7a54fa9adc/sdk/typespec/typespec_client_core/examples/stream_response.rs#L20-L27